StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1000CodeFixProvider.cs

```diff
@@ -88,25 +89,27 @@ public override FixAllProvider GetFixAllProvider()
                     break;
                 }
 
-                if (isAddingSpace)
+                if (isAddingSpace == !token.HasTrailingTrivia)
                 {
-                    if (token.HasTrailingTrivia)
-                        continue;
-
-                    SyntaxTrivia whitespace = SyntaxFactory.Whitespace(" ").WithoutFormatting();
-                    SyntaxToken corrected = token.WithTrailingTrivia(token.TrailingTrivia.Insert(0, whitespace));
-                    Document updatedDocument = context.Document.WithSyntaxRoot(root.ReplaceToken(token, corrected));
-                    context.RegisterCodeFix(CodeAction.Create("Fix spacing", t => Task.FromResult(updatedDocument)), diagnostic);
+                    context.RegisterCodeFix(CodeAction.Create("Fix spacing", t => GetTransformedDocument(context, root, token, isAddingSpace)), diagnostic);
```

Comment:
:exclamation: Move as much of the logic from this loop to the callback as possible. Most of the work is still being done in the main loop.